src/core/api/notification_policy.go

```diff
@@ -128,6 +129,11 @@ func (w *NotificationPolicyAPI) Post() {
 
 	id, err := notification.PolicyMgr.Create(policy)
 	if err != nil {
+		if e := orm.AsConflictError(err, "notification policy named %s already exists", policy.Name); e != nil {
```

Comment:
we should put the logic in the dao level, like
https://github.com/goharbor/harbor/blob/master/src/pkg/label/dao.go#L79

Reply:
considering the 409 / 500 response code should be sent in different circumstances

Reply:
You can use the `SendError` method which will return the response code depending on the err.

Reply:
fixed